addons/point_of_sale/models/pos_session.py

```diff
@@ -278,15 +282,468 @@ def _validate_session(self):
 
     def action_pos_session_close(self):
-        # Close CashBox
-        for st in self.statement_ids:
-            if any(abs(st.difference) > self.config_id.amount_authorized_diff for st in self.statement_ids):
-                if not self.user_has_groups("point_of_sale.group_pos_manager"):
-                    raise UserError(_("Your ending balance is too different from the theoretical cash closing (%.2f), the maximum allowed is: %.2f. You can contact your manager to force it.") % (st.difference, self.config_id.amount_authorized_diff))
+        # Session without cash payment method will not have a cash register.
+        # However, there could be other payment methods, thus, session still
+        # needs to be validated.
+        if not self.cash_register_id:
+            return self._validate_session()
+
+        if self.cash_control and abs(self.cash_register_difference) > self.config_id.amount_authorized_diff:
+            # Only pos manager can close statements with cash_register_difference greater than amount_authorized_diff.
+            if not self.user_has_groups("point_of_sale.group_pos_manager"):
+                raise UserError(_(
+                    "Your ending balance is too different from the theoretical cash closing (%.2f), "
+                    "the maximum allowed is: %.2f. You can contact your manager to force it."
+                ) % (self.cash_register_difference, self.config_id.amount_authorized_diff))
+            else:
+                return self._warning_balance_closing()
+        else:
+            return self._validate_session()
+
+    def _create_account_move(self):
+        """ Create account.move and account.move.line records for this session.
+
+        Side-effects include:
+            - setting self.move_id to the created account.move record
+            - creating and validating account.bank.statement for cash payments
+            - reconciling cash receivable lines, invoice receivable lines and stock output lines
+        """
+        journal = self.config_id.journal_id
+        # Passing default_journal_id for the calculation of default currency of account move
+        # See _get_default_currency in the account/account_move.py.
+        account_move = self.env['account.move'].with_context(default_journal_id=journal.id).create({
+            'journal_id': journal.id,
+            'date': fields.Date.context_today(self),
+            'ref': self.name,
+            'name': journal.sequence_id.next_by_id()
+        })
+        self.write({'move_id': account_move.id})
+
+        ## SECTION: Accumulate the amounts for each accounting lines group
+        # Each dict maps `key` -> `amounts`, where `key` is the group key.
+        # E.g. `combine_receivables` is derived from pos.payment records
+        # in the self.order_ids with group key of the `payment_method_id`
+        # field of the pos.payment record.
+        split_receivables = defaultdict(lambda: {'amount': 0.0, 'amount_converted': 0.0})
```

Comment:
Istead of default dict, could you symply do something like this:
amounts = {'amount': 0.0, 'amount_converted': 0.0}
split_receivables = dict(**amounts)
taxes = dict(**amounts, base_amount=0.0)
....

Reply:
I need the defaultdict so that if I access a key that doesn't exist, then I will have a default value of `{'amount': 0.0, 'amount_converted': 0.0}`. It is essential in incrementing the amount for each key.

e.g. the `invoice_receivables` - its values are grouped by receivable account of the customer. 
Considering the following orders:
- order 1 amount = 10, receivable account = 10
- order 2 amount = 12, receivable account = 6
- order 3 amount = 13,  receivable account = 10

looping to the orders will result:
order 1
invoice_receivables[10] := {'amount': 0.0, 'amount_converted': 0.0}
  after `update amounts`
invoice_receivable[10] := {'amount': 10, 'amount_converted': 10}

order 2
invoice_receivables[6] :=  {'amount': 0.0, 'amount_converted': 0.0}
  after `update amounts`
invoice_receivable[6] := {'amount': 12, 'amount_converted': 12}

order 3
invoice_receivables[10] := {'amount': 10, 'amount_converted': 10}
  after `update amounts`
invoice_receivable[10] := {'amount': 23, 'amount_converted': 23}